spark/src/test/scala/org/apache/spark/sql/delta/perf/OptimizeMetadataOnlyDeltaQuerySuite.scala

```diff
@@ -188,99 +640,261 @@ class OptimizeMetadataOnlyDeltaQuerySuite
+    val tableName = "TestNestedColumns"
+
+    spark.sql(s"CREATE TABLE $tableName " +
+      s"(Column1 STRUCT<Id: INT>, " +
+      s"`Column1.Id` INT) USING DELTA")
+
+    spark.sql(s"INSERT INTO $tableName" +
+      s" (Column1, `Column1.Id`) VALUES " +
+      s"(STRUCT(1), 2);")
+
+    // Nested Column
+    checkOptimizationIsNotTriggered(
+      s"SELECT MAX(Column1.Id) FROM $tableName")
+
+    checkOptimizationIsNotTriggered(
+      s"SELECT MAX(Column1.Id) AS XYZ FROM $tableName")
+
+    // Validate the scenario where all the columns are read
+    // since it creates a different query plan
+    checkOptimizationIsNotTriggered(
+      s"SELECT MAX(Column1.Id), " +
+        s"MAX(`Column1.Id`) FROM $tableName")
+
+    // The optimization for columns with dots should still work
+    checkResultsAndOptimizedPlan(s"SELECT MAX(`Column1.Id`) FROM $tableName",
+      "LocalRelation [none#0]")
   }
 
-  test("Select Count: group by") {
-    checkSameQueryPlanAndResults(
-      s"SELECT group, COUNT(*) FROM $testTableName GROUP BY group",
-      Seq(Row(1L, totalRows)))
+  test("count-min-max - not supported - group by") {
+    checkOptimizationIsNotTriggered(
+      s"SELECT group, COUNT(*) FROM $testTableName GROUP BY group")
+
+    checkOptimizationIsNotTriggered(
+      s"SELECT group, MIN(id), MAX(id) FROM $testTableName GROUP BY group")
   }
 
-  test("Select Count: count twice") {
-    checkSameQueryPlanAndResults(
-      s"SELECT COUNT(*), COUNT(*) FROM $testTableName",
-      Seq(Row(totalRows, totalRows)))
+  test("count-min-max - not supported - plus literal") {
+    checkOptimizationIsNotTriggered(
+      s"SELECT COUNT(*) + 1 FROM $testTableName")
+
+    checkOptimizationIsNotTriggered(
+      s"SELECT MAX(id) + 1 FROM $testTableName")
   }
 
-  test("Select Count: plus literal") {
-    checkSameQueryPlanAndResults(
-      s"SELECT COUNT(*) + 1 FROM $testTableName",
-      Seq(Row(totalRows + 1)))
+  test("count - not supported - distinct count") {
+    checkOptimizationIsNotTriggered(
+      s"SELECT COUNT(DISTINCT data) FROM $testTableName")
   }
 
-  test("Select Count: distinct") {
-    checkSameQueryPlanAndResults(
-      s"SELECT COUNT(DISTINCT data) FROM $testTableName",
-      Seq(Row(totalDistinctData)))
+  test("count-min-max - not supported - filter") {
+    checkOptimizationIsNotTriggered(
+      s"SELECT COUNT(*) FROM $testTableName WHERE id > 0")
+
+    checkOptimizationIsNotTriggered(
+      s"SELECT MAX(id) FROM $testTableName WHERE id > 0")
   }
 
-  test("Select Count: filter") {
-    checkSameQueryPlanAndResults(
-      s"SELECT COUNT(*) FROM $testTableName WHERE id > 0",
-      Seq(Row(totalRows)))
+  test("count-min-max - not supported - sub-query with filter") {
+    checkOptimizationIsNotTriggered(
+      s"SELECT (SELECT COUNT(*) FROM $testTableName WHERE id > 0)")
+
+    checkOptimizationIsNotTriggered(
+      s"SELECT (SELECT MAX(id) FROM $testTableName WHERE id > 0)")
```

Comment:
test case when the filter is both on data and partition columns?

Reply:
Also is there is a case where the filter is just on partition columns?

Reply:
Added